tests/codegen/simd-intrinsic/simd-intrinsic-transmute-array.rs

```diff
@@ -21,20 +21,48 @@ pub struct U(f32, f32, f32, f32);
 // CHECK-LABEL: @build_array_s
 #[no_mangle]
 pub fn build_array_s(x: [f32; 4]) -> S<4> {
-    // CHECK: call void @llvm.memcpy.{{.+}}({{.*}}, i{{[0-9]+}} 16, i1 false)
+    // CHECK: call void @llvm.memcpy.{{.+}}({{.*}} align 16 {{.*}} align 4 {{.*}} 16, i1 false)
     S::<4>(x)
 }
 
+// CHECK-LABEL: @build_array_transmute_s
+#[no_mangle]
+pub fn build_array_transmute_s(x: [f32; 4]) -> S<4> {
+    // CHECK: %[[VAL:.+]] = load <4 x float>, {{ptr %x|.+>\* %.+}}, align 4
+    // CHECK: store <4 x float> %[[VAL:.+]], {{ptr %0|.+>\* %.+}}, align 16
+    unsafe { std::mem::transmute(x) }
```

Comment:
Don't think it's a problem, but it's interesting how `S(x)` produces a `memcpy` while transmute produces load+store, while doing identical work (I think?).

Reply:
Quite!  When I saw this I [brought it up with the portable-simd folks](https://rust-lang.zulipchat.com/#narrow/stream/257879-project-portable-simd/topic/.60repr.28simd.29.60.20inconsistency/near/346252120), since it seems that LLVM doesn't normalize it in IR in either direction, so they might want to experiment with which is better.